slashing-protection/src/test/java/tech/pegasys/web3signer/slashingprotection/validator/GenesisValidatorRootValidatorTest.java

```diff
@@ -23,22 +26,46 @@
 import org.jdbi.v3.core.Jdbi;
 import org.junit.jupiter.api.Test;
 import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mockito;
+import org.mockito.Spy;
+import org.mockito.junit.jupiter.MockitoExtension;
 
 @ExtendWith(DatabaseSetupExtension.class)
+@ExtendWith(MockitoExtension.class)
 class GenesisValidatorRootValidatorTest {
+  @Spy private MetadataDao metadataDao = new MetadataDao();
+
   @Test
   void checkGenesisValidatorReturnTrueForNewGVR(final Jdbi jdbi) {
     final GenesisValidatorRootValidator gvrValidator =
-        new GenesisValidatorRootValidator(jdbi, new MetadataDao());
-    assertThat(
-            gvrValidator.checkGenesisValidatorsRootAndInsertIfEmpty(Bytes32.leftPad(Bytes.of(3))))
+        new GenesisValidatorRootValidator(jdbi, metadataDao);
+    final Bytes32 genesisValidatorsRoot = Bytes32.leftPad(Bytes.of(3));
+    assertThat(gvrValidator.checkGenesisValidatorsRootAndInsertIfEmpty(genesisValidatorsRoot))
         .isTrue();
   }
 
```

Comment:
Also be good to add a test that shows that is it not cached when we fail insert the GVR due to a database exception

Reply:
done